File: 1984. Minimum Difference Between Highest and Lowest of K Scores/minimumDifference.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        if(k == 1)return 0;
        sort(nums.begin(), nums.end(), greater<int>());
        int ans = 100001;

        int j = k-1;

        for(int i=0; j<nums.size(); i++,j++){
            ans = min(ans, nums[i]-nums[j]);
        }

        return ans;
    }
};

File: 1984. Minimum Difference Between Highest and Lowest of K Scores/minimumDifference.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function (nums, k) {
    if (k === 1) return 0
    nums.sort((a, b) => b - a)
    let ans = Infinity
    let j = k - 1 //the last item in section we looked in

    for (let i = 0; j < nums.length; i++, j++) {
        ans = Math.min(ans, nums[i] - nums[j])
    }
    return ans;
};